src/section/section_map.rs

```diff
@@ -0,0 +1,461 @@
+    pub fn compatible(&self, prefix: &Prefix<XorName>) -> Option<&EldersInfo> {
+        self.all()
+            .find(move |(pfx, _)| pfx.is_compatible(prefix))
+            .map(|(_, info)| info)
+    }
+
+    /// Find other section containing the given elder.
+    pub fn find_other_by_member(&self, pub_id: &PublicId) -> Option<&EldersInfo> {
+        self.other
+            .iter()
+            .find(|(_, info)| info.is_member(pub_id))
+            .map(|(_, info)| info)
+    }
+
+    /// Returns the known section that is closest to the given name, regardless of whether `name`
+    /// belongs in that section or not.
+    pub fn closest(&self, name: &XorName) -> (&Prefix<XorName>, &EldersInfo) {
+        let mut best_pfx = self.our().prefix();
+        let mut best_info = self.our();
+        for (pfx, info) in self.all() {
+            // TODO: Remove the first check after verifying that section infos are never empty.
+            if !info.is_empty() && best_pfx.cmp_distance(pfx, name) == Ordering::Greater {
+                best_pfx = pfx;
+                best_info = info;
+            }
+        }
+
+        (best_pfx, best_info)
+    }
+
+    /// Returns iterator over all known sections.
+    pub fn all(&self) -> impl Iterator<Item = (&Prefix<XorName>, &EldersInfo)> + Clone {
+        iter::once((self.our.last().prefix(), self.our.last())).chain(&self.other)
+    }
+
+    /// Returns iterator over all known sections excluding ours.
+    pub fn other(&self) -> impl Iterator<Item = (&Prefix<XorName>, &EldersInfo)> {
+        self.other.iter()
+    }
+
+    /// Returns the known sections sorted by the distance from a given XorName.
+    pub fn sorted_by_distance_to(&self, name: &XorName) -> Vec<(&Prefix<XorName>, &EldersInfo)> {
+        let mut result: Vec<_> = self.all().collect();
+        result.sort_by(|lhs, rhs| lhs.0.cmp_distance(rhs.0, name));
+        result
+    }
+
+    /// Returns prefixes of all known sections.
+    pub fn prefixes(&self) -> impl Iterator<Item = &Prefix<XorName>> + Clone {
+        self.all().map(|(prefix, _)| prefix)
+    }
+
+    /// Returns `true` if the `EldersInfo` isn't known to us yet.
+    pub fn is_new(&self, elders_info: &EldersInfo) -> bool {
+        !self.all().any(|(_, info)| info.is_newer(elders_info))
+    }
+
+    /// Returns `true` if the `EldersInfo` isn't known to us yet and is a neighbouring section.
+    pub fn is_new_neighbour(&self, elders_info: &EldersInfo) -> bool {
+        let our_prefix = self.our().prefix();
+        let other_prefix = elders_info.prefix();
+
+        (our_prefix.is_neighbour(other_prefix) || other_prefix.is_extension_of(our_prefix))
+            && self.is_new(elders_info)
+    }
+
+    /// Returns all elders from all known sections.
+    pub fn elders(&self) -> impl Iterator<Item = &P2pNode> {
+        self.all()
+            .map(|(_, info)| info)
+            .flat_map(EldersInfo::member_nodes)
```

Comment:
I always feel bit confused with this `elders` and `member` mismatch   :(
same comment to the others.

Reply:
Perhaps 
members == all 
mature == all except infants 
Then Infant, Adults, Elders

Reply:
Yeah, that will be more explicit I think.

Reply:
Yeah, it's because the `EldersInfo` struct contains only elders, so `member` means actually elder. But I can see the confusion and will change it.